libut/ut.c

```diff
@@ -747,7 +747,11 @@ ut_generate(URITEMP ut, const char *template, char *out, size_t outlen)
 			}
 
 			vlist = strdup(p);
-			vlist[vlistlen - 1] = '\0';
+			/* Prevent integer underflow: ensure vlistlen > 0 before subtracting */
+			if (vlistlen > 0 && vlistlen <= strlen(vlist))
+			{
+				vlist[vlistlen - 1] = '\0';
+			}
 
```

Comment:
`vlistlen` here  always  `>= 1` because  `p` should be NULL terminated string and `*p` points other than `\0` (`}` is not found while scanning).  And then vlist is never never truncated as I wrote [the comment on PR #3](https://github.com/futatuki/OpenDKIM/pull/3/files#r2642505226), I think.